internal/setup/setup.go

```diff
@@ -458,12 +458,23 @@ func parseRelease(baseDir, filePath string, data []byte) (*Release, error) {
 		if details.Default {
 			release.DefaultArchive = archiveName
 		}
+		if len(details.PublicKeys) == 0 {
+			return nil, fmt.Errorf("%s: archive %q missing public-keys field", fileName, archiveName)
+		}
+		var archiveKeys []*packet.PublicKey
+		for _, keyName := range details.PublicKeys {
+			key, ok := pubKeys[keyName]
+			if !ok {
+				return nil, fmt.Errorf("%s: unknown reference to public key %q in archive %q", fileName, keyName, archiveName)
```

Comment:
`%s: archive %q refers to undefined public key %q`

Besides being more compact, in general when we can phrase it in the order of hierarchy, it's easier to read and understand the error. I won't typically be bothering you with these kinds of issues, though, but since you're getting started on this project it's worth keeping in mind.

Reply:
It is indeed helpful to create a shared understanding of how to present the errors to the user.